worker.go

```diff
@@ -248,13 +240,11 @@ If an error on a correct message that has been redelivered ie failed before, the
 If an error on a incorrect message is returned, the message is not requeued but nacked (this would cause the queue to fill up)
 If no error is returned, the message is acked
 */
-func handleMessages(messages <-chan amqp.Delivery, handler AmqpHandler, done chan error) {
+func handleMessages(messages <-chan amqp.Delivery, handler AmqpHandler, waitGroup *sync.WaitGroup) {
+	defer waitGroup.Done()
```

Comment:
it would be even better to wait for all messages to be handled properly before leaving.
@gcoupelant @jean-k-cp What is your advice?

Reply:
Fixed. I'll ask if a timeout is needed or not

Reply:
As discussed with @jean-k-cp, stakhanov waits a little bit before closing the channels. Since we have no guarantee about when a handler will stop. This is what I'll do.